araproc/analysis/snr.py

```diff
@@ -96,6 +97,77 @@ def get_min_segmented_rms(waveform, nsegs=8):
 
     return rms
 
+def get_jackknife_rms(waveform, nSamples=50):
+
+    """
+    Calculates the RMS of a voltage trace via a sort-of jackknife technique,
+    which attempts to identify the parts of the trace which do not contain 
+    signal with the goal of using as much of the trace to estimate the RMS 
+    as possible. The RMS is then calculated using all parts of the trace which
+    are determined to not be contain signal (ie outlier voltages). The trace
+    is segmented into equal-sample segments to avoid sensitivity to the trace
+    length.
+
+    Parameters
+    ----------
+    waveform: TGraph or np.ndarray
+        A TGraph or np.ndarray of the waveform voltage.
+    nSamples: int
+        Number of samples for each segment.
+
+    Returns
+    -------
+    rms : float
+        The jackknifed RMS in same units as trace.
+    """
+    
+    if(isinstance(waveform, ROOT.TGraph)):
+      _, trace = wfu.tgraph_to_arrays(waveform)
+    elif(isinstance(waveform, np.ndarray)):
+      trace = np.copy(waveform)
+      
+      # don't be bothered by some unused dimensions
+      trace = np.squeeze(trace)
+
+      if(trace.ndim != 1):
+        raise Exception("Trace is not 1d in snr.get_jackknife_rms. Abort")
+
+    else:
+      raise Exception("Unsupported data type in snr.get_jackknife_rms. Abort")
+
+    # first split the trace into segments of nSamples 
+    # then calculate the RMS on the trace _without_ that segment (ie the "all-but RMS")
+    allRms = []
```

Comment:
```suggestion
    # Split the trace into segments
    segments = [trace[i:i + nSamples] for i in range(0, len(trace), nSamples)]

    # Calculate the all-but RMS for each segment
    allRms = []
    for i in range(len(segments)):
        subTrace = np.concatenate(segments[:i] + segments[i + 1:])
        allRms.append(np.sqrt(np.mean(subTrace**2)))

    allRms = np.array(allRms)
```

something like this works better?